### ISSUE TITLE
Add option to TreeNode to enable button behaviour with key modifiers enabled.

### ISSUE THREAD

**Issue by AlvaroBarua** [ocornut/imgui#2886]
Hi, I am working on a multiselection tree hierarchy and using LCTRL as the modifier to select multiple items. My idea is to be able to while holding down the LCTRL key, open a tree node if I click on the arrow. Right now, the default behaviour of the button behaviour inside the TreeNode is to include the ImGuiButtonFlags_NoKeyModifiers flag which prevents me from doing so. 

I have manually removed the default flag, but I guess having a proper way of disabling this behaviour would be the ideal solution.

Thanks.
```
\\Won't change state when LCTRL is held
bool treeNodeOpen = ImGui::TreeNodeEx(name, treeNodeFlasg);
bool clickedOnArrow = (ImGui::GetMousePos().x - ImGui::GetItemRectMin().x) < ImGui::GetTreeNodeToLabelSpacing();

if (ImGui::IsItemClicked())
{
    if (Input::GetKey(KeyCode::LCTRL) && !clickedOnArrow)
   {
       //Add item to list
   }
}

if (treeNodeOpen)
    RecursiveBuild(...);

```

**Comment by ocornut** [comment]
Which tree node flags are you using? Are you using `ImGuiTreeNodeFlags_OpenOnArrow`, or are not your using it and expecting a different behavior based on whether CTRL and click are happening over the arrow or before/after ?

My gut feeling is that for tree node it would make sense to allow toggling the node open state with modifiers when clicking specifically on the arrow, so we could alter the behavior based on mouse position. I'll have to check with `features/range_select` and think about other side-effects or doing this.

Potential fix would be to move the arrow_x1/arrow_x2 declaration higher in the function and alter the button flags based on `io.MousePos.x`.

(Also need to consider `ImGuiTreeNodeFlags_SpanFullWidth` being used which mean we can click _before_ the arrow.)

Also consider looking at #1861 and the `features/range_select` branch.

(PS: Please fill the issue template as requested when creating a new issue.)

**Comment by ocornut** [comment]
I applied the change described above now.

In addition, I have added `IsItemToggledOpen()` to explicitely query if the node was just open/closed which is required to fully handle #1896.

**Comment by AlvaroBarua** [comment]
Thank you for the quick fix, it is exactly what I needed.

I was doing a range check to see if the mouse was in the arrow area when clicked using the TreeDepth, but with IsItemToggledOpen() should be way easier and cleaner.

And sorry for the template, I must have cleared the text and forgot about it completely.
